### ISSUE TITLE
`@envelop/generic-auth`: docs and implementation are different

### ISSUE THREAD

**Issue by taneba** [graphql-hive/envelop#614]
**Describe the bug**

There seems to be several differences between docs and implementation.

1. @envelop/generic-auth has no exported member named 'resolveUser'.

in the docs:

```
import { useGenericAuth, resolveUser, ValidateUserFn } from '@envelop/generic-auth';
```

But `resolveUser` does not exist. So this should be ResolveUserFn.

2. resolveUser does not exist in the type of GenericAuthPluginOptions

in the docs:

```
    useGenericAuth({
      resolveUser,
      validateUser,
      mode: 'protect-all',
    }),
```

But resolveUser does not exist in the type of GenericAuthPluginOptions as I saw implementation.
Here, `resolveUserFn: ResolveUserFn<UserType, ContextType>;` and `validateUser?: ValidateUserFn<UserType, ContextType>;` are different in naming convention. I think it's better to use same naming rule in terms of consistency.

**Environment:**

- "@envelop/generic-auth": "^1.0.0"
- "@envelop/core": "^1.0.3"
- NodeJS:

**Comment by taneba** [comment]
IMO resolverUser seems better than resolveUserFn because -Fn suffix looks more like a type definition. 
If this makes sense, I'll send PR to rename and fix docs :)

**Comment by taneba** [comment]
Additionally, directiveNode seems not to be passed to validateUserFn on execute.

**Comment by dotansimha** [comment]
Fixed in `@envelop/generic-auth@1.0.1`. Thanks @taneba
